### PR TITLE
METAMODEL-1151: Fixed by adding DataContextFactory for SugarCRM.

### PR DESCRIPTION
**Pull request by kaspersorensen**
As per https://issues.apache.org/jira/browse/METAMODEL-1151 only SugarCRM was missing a DataContextFactory, so I added it :-)

### CODE REVIEW

File: sugarcrm/src/main/java/org/apache/metamodel/sugarcrm/SugarCrmDataContextFactory.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+public class SugarCrmDataContextFactory extends AbstractDataContextFactory {
+
+    @Override
+    protected String getType() {
+        return "sugarcrm";
+    }
+
+    @Override
+    public DataContext create(DataContextProperties properties, ResourceFactoryRegistry resourceFactoryRegistry)
```
Comment (LosD) [inline sugarcrm/src/main/java/org/apache/metamodel/sugarcrm/SugarCrmDataContextFactory.java]: Can't remember if we have "rules" here, but why not `final`? Arguments are just as susceptible to being manipulated as variables, and can be just as surprising (if not more).

(of course, in as simple a function as this, it doesn't really matter for any of them)

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ### Apache MetaModel _WIP_
 
+ * [METAMODEL-1151] - Created DataContextFactory implementations for earch remaining module.
```
Comment (LosD) [inline CHANGES.md]: Typo, earch => each

File: dynamodb/src/main/java/org/apache/metamodel/dynamodb/DynamoDbDataContextFactory.java [hunk 3]
```diff
@@ -31,6 +32,9 @@
 import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
 import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
 
+/**
+ * {@link DataContextFactory} for DynamoDB
```
Comment (LosD) [inline dynamodb/src/main/java/org/apache/metamodel/dynamodb/DynamoDbDataContextFactory.java]: A liiiiittle bit useless JavaDoc comment. :)

**Comment by kaspersorensen** [comment]
Thanks for the review @LosD, I love that you're obnoxiously pedantic and detail oriented ;-) I corrected the PR as per your feedback items.

**Comment by LosD** [comment]
It's what I do :smile:

LGTM! Will merge when I'm back at my computer (you can of course also just do it now if you prefer)

**Pull request closed without merging**